tensilelite/Tensile/KernelWriter.py

```diff
@@ -3971,23 +4011,29 @@ def readWriteVectors(mat, vw, kernel):
     self.states.a.numVgprLocalWriteSwapAddr = 0
     self.states.b.numVgprLocalWriteSwapAddr = 0
     self.states.m.numVgprLocalWriteSwapAddr = 0
+    self.states.a.numVgprLocalWriteAddrTailLoop = 0 if not (kernel["DirectToLdsA"] and kernel["NonDTLTailLoop"]) else 1 * self.states.rpla
+    self.states.b.numVgprLocalWriteAddrTailLoop = 0 if not (kernel["DirectToLdsB"] and kernel["NonDTLTailLoop"]) else 1 * self.states.rpla
 
     if self.states.archCaps["DeviceLDS"] > 65536 and not kernel["StoreSwapAddr"]:
       need128K = kernel["LdsOffsetA_Blk"]>=131072 and kernel["ExpandPointerSwap"] and not kernel["1LDSBuffer"]
       need64K = kernel["LdsOffsetA_Blk"]>=65536 and kernel["ExpandPointerSwap"] and not kernel["1LDSBuffer"]
       if need128K or kernel["LdsNumElementsAlignedA"]>=131072:
         self.states.a.numVgprLocalReadAddr *= 3
         self.states.a.numVgprLocalWriteAddr *= 3
+        self.states.a.numVgprLocalWriteAddrTailLoop *= 3
```

Comment:
Not necessary to do this in this PR, but I suggest to refactor this part.
- replace 65536 related const values with offset limit
- calculate num of multiple factor (*3 or *2) before setting numVgprLocalRead/Write addr. Then, use the num of multiple factor when setting numVgprLocalRead/Write addr (replace 1* with num *)

This part looks redundant.
I suggest to simplify it instead of adding *3 and *2 for new parameters.